app.py

```diff
@@ -0,0 +1,74 @@
+from flask import Flask, render_template, request, redirect, url_for, jsonify, flash
+import mysql.connector
+
+app = Flask(__name__)
+app.secret_key = '\xbd\xa4\xe9_4[!\xe5oW\xc9\xfc\xdfM+\xaf\x9b\x88my\xd6\xfe\xd9T'  # Replace with your own secret key
+
+# Database connection
+db = mysql.connector.connect(
+    host='localhost',
+    user='root',
+    password='Bpr123',
+    database='mysqll'
+)
+cursor = db.cursor()
+
+@app.route('/')
+def index():
+      # Retrieve all tasks from the database
+        cursor.execute('SELECT * FROM tasks')
+        tasks = cursor.fetchall()
```

Comment:
Similarly getting all the Tasks become so easy and even you can handle the 404 in better way. Tasks class gives you many options.

```
@app.route('/tasks/<int:task_id>', methods=['GET'])
def get_task(task_id):
    task = Task.query.get_or_404(task_id)
    return jsonify({"id": task.id, "title": task.title, "description": task.description, "complete": task.complete})

```